### ISSUE TITLE
Iridium on the Add-on store might not receive any further updates

### ISSUE THREAD

**Issue by ParticleCore** [ParticleCore/Iridium#828]
Version 1.2.5 was released and submitted for the AMO public listing. This fix is actually a workaround for a bug that Firefox has yet to fix ([read more here](https://github.com/ParticleCore/Iridium/issues/825)).
In the update nothing else was added or modified, the [workaround was the only thing implemented](https://github.com/ParticleCore/Iridium/commit/f051f90f6103c296d8729ee3737ea94fa77dbd7a)
A reviewer decided to ask for the add-on to implement a prompt for disclosing the use of the cookie permission, quoting:

> 1) The user must be provided an opportunity to refuse the storage of or access to cookies, and must be informed of the consequences of doing so (e.g., without a functional cookie, the add-on may not work)

Not only the user has to be told the reason for the use of the permission (which is already disclaimed in the AMO add-on description), but the add-on would also need to allow the user NOT authorize the use of that permission, EXPLAIN what will happen if the user does so and let the developer deal with whatever consequence that causes to the add-on.

I have very limited time at the moment to dedicate to the add-on, I've been always doing my best to at lest keep it working when something breaks and I have endured countless retarded (there is no other way of saying this) demands by the AMO reviewers. I've had one reviewer once demanding for me to undo all the changes that a content script does on any page it runs on. For those of you that understand what this means, you also understand why I use the word "retarded".

I'm tired, I'm busy and I have no time to waste any more on this kind of circus. If the add-on on the AMO store does not receive any more updates that means they insist on being unreasonable and I will no longer tolerate that.

The unlisted version is exactly the same, is updated and also receives updates faster. If you wish to continue receiving updates you can find the most recent version here: https://github.com/ParticleCore/Iridium/wiki/Download

I am sad that it comes down to this since I was personally asked to return to the AMO store and I thought I'd give it one last shot, but I have a life and my health to prioritize.

**Comment by thepcwiz101** [comment]
This is one change that i find to be very stupid that firefox implements. Some devs aren't able to change their addon's every time firefox or even google chrome changes things.

**Comment by karma12gaming** [comment]
I'd rather time be dedicated to development instead of fighting the store so makes sense to me.

**Comment by Jdbye** [comment]
Permission management is maybe something that should be implemented at the browser level itself, much like how Google has done it with later Android versions. No sense in reimplementing the wheel for every addon created if this is something they're going to require of everyone.

That being said, I also understand that not every user will be comfortable with an addon having access to their cookies and if the option can't be turned off will just choose not to use the addon. So it wouldn't be a bad idea to have a toggle switch for that feature, if it's not something essential to the addon functionality (generally speaking, many people might not need those couple of features that granting a certain permission enables and would be just fine turning the permission off)
But if the addon is managing its own permissions, how can you trust the developer that they don't collect cookies in the background (hypothetically speaking) even when the feature is turned off? Permissions have to be implemented at the browser level for them to actually be meaningful.
So while I think it's a nice idea their suggestion of implementing it at the addon level makes no sense to me.

**Comment by ParticleCore** [comment]
@Jdbye I agree with everything you said, it is exactly like that. I hope they implement it in the browser and when that day comes I will gladly update the add-on to comply with the new changes and handle each permission user control. Until then it will stay as it is.

**Comment by ParticleCore** [comment]
Main topic has been updated with new information.

**Comment by ParticleCore** [comment]
The topic has been updated with the final information.

**Comment by gluons** [comment]
@ParticleCore You should create a simple page (maybe on [GitHub Pages](https://pages.github.com/)) with a link to `.xpi` file.  
Firefox will ask to install add-on. So, it may simplify installation steps.

![ภาพ](https://user-images.githubusercontent.com/4688092/78249829-2d212300-7519-11ea-8dfb-5c36df9dd80c.png)

![ภาพ](https://user-images.githubusercontent.com/4688092/78249892-44f8a700-7519-11ea-92e2-48b0485e0e39.png)

**Comment by ParticleCore** [comment]
@gluons Thanks for the suggestion. That is correct, GitHub serves the link with an incorrect content type (or correct, depending on the point of view), which is why I've included the instructions.

I don't know if serving the file through the GH pages will make the content type work correctly, but worth a short for sure so I will check it and if it works I will update the wiki with the new link.

**Comment by gluons** [comment]
I've created experimental download page (https://gluons.gitlab.io/iridium-download-page/).  
Not sure whether it'll work or not.

**Comment by amjd** [comment]
@gluons I just tried it. It works perfectly. Also, cool trick of using jsDelivr. Good job! :)

**Comment by NovaViper** [comment]
Will there ever be a version for Chrome?

**Comment by ParticleCore** [comment]
> 
> 
> Will there ever be a version for Chrome?

When Chrome implements the API, until then it's a no since the API just doesn't exist in Chrome.

**Comment by octapusxft** [comment]
@gluons 
Is it just my mistaken memory or is the Iridium version on the download page missing some features?
Wasn't iridium able to reverse the YT playlist order?
In any case thanks for your efforts! I could not manually install the addon because it was giving a corrupted error

**Comment by gluons** [comment]
@octapusxft 

> In any case thanks for your efforts! I could not manually install the addon because it was giving a corrupted error

Have you tried installing via [manual method](https://github.com/ParticleCore/Iridium/wiki/Download)? Does it work?  
My site is experimental. I'm not sure if it'll work perfectly.

---

> Is it just my mistaken memory or is the Iridium version on the download page missing some features?
Wasn't iridium able to reverse the YT playlist order?

If you can install Iridium correct and still don't see your expected features, maybe it's because of new version ([Starting from scratch](https://github.com/ParticleCore/Iridium/releases/tag/1.0.0)).

Or you can ask @ParticleCore about features (maybe in the other new issue?) after you can install Iridium correctly.

**Comment by octapusxft** [comment]
@gluons 
The manual install gives me an error that the files are corrupted and wont proceed. I do not know why.
Anyway I will try not bothering ParticleCore for now I will try again in the future and if I have issues I will make a bug report.

**Comment by ParticleCore** [comment]
Remember to uninstall previous versions before installing a new one if you intend to switch from the AMO store hosted version to the GitHub hosted version.

**Comment by electrofloat** [comment]
Anyone care to explain what the f* is 'AMO'?

**Comment by gluons** [comment]
`AMO` stands for `addons.mozilla.org`.
